### ISSUE TITLE
feat: New select component types

### ISSUE THREAD

**Issue by sudojunior** [Snazzah/slash-create#264]
Recent changes on the datamine reveal additional select component types for mentionable targets.
https://github.com/Discord-Datamining/Discord-Datamining/commit/4dbc3575cf7b026953c3767a425c849bdb949388#commitcomment-68333268

3. `SELECT` -> `STRING_SELECT`
---
5. `USER_SELECT`
6. `ROLE_SELECT`
7. `MENTIONABLE_SELECT`
8. `CHANNEL_SELECT`

Structure of these new components remains to be said, but will likely follow a similar structure for the existing enum on `CommandOptionType`. Would expect `#values` to be a list of IDs, and then have user objects found in the [`#resolved` Data Structure](https://discord.com/developers/docs/interactions/receiving-and-responding#interaction-object-resolved-data-structure).

**Comment by sudojunior** [comment]
With what I'd say is a 90% / 95% accuracy, I believe a `BaseInteractionContext` will be _possible_ that shares declarations between all available contexts (except PING). `user`, `interactionID`, `interactionToken` and `invokedAt` (as a default for expiry) are all guaranteed to exist on these contexts.

I have created a [spreadsheet (GSheets)](https://docs.google.com/spreadsheets/d/1gNh0u1mpvTHxt2zE-RMQ-Izc_lHZ0OjsR7eQVPVkwck/edit?usp=sharing) to illustrate the member availability across all contexts, and where I see this heading in the immediate sense. _This is, of course, likely to change with how Discord themselves handle this and subsequently how `slash-create` would adapt to fit the new parameters._
> At this time, `AutocompleteContext` and `MessageInteractionContext` share the most duplicated member references at 9 occurrences from my observations.

In the past, slash-create has been ahead of competitor libraries on how Application Commands were implemented - which resulted in disparity within deployments and how they handled newer structures. I would expect this to happen again, if Discord is as swift as they have been for new or updated features prior to this.
